crates/relayer/src/chain/endpoint.rs

```diff
@@ -91,7 +94,19 @@ pub trait ChainEndpoint: Sized {
     fn init_event_monitor(
         &self,
         rt: Arc<TokioRuntime>,
-    ) -> Result<(EventReceiver, TxMonitorCmd), Error>;
+    ) -> Result<(EventReceiver, TxMonitorCmd), Error> {
+        crate::time!("init_event_monitor");
+
+        let (mut event_monitor, event_receiver, monitor_tx) =
+            EventMonitor::new(self.id().clone(), self.config().websocket_addr.clone(), rt)
+                .map_err(Error::event_monitor)?;
+
+        event_monitor.subscribe().map_err(Error::event_monitor)?;
+
+        thread::spawn(move || event_monitor.run());
```

Comment:
Curious to hear @romac weigh in on the implications (if any) of moving the thread spawning + subscription here in the endpoint trait directly. I don't see any negatives.